### PR TITLE
Fix some issues with edit redeclerable parameter button

### PR DESCRIPTION
**Pull request by MaksKBV**
### Related Issues

Incorrect component name in dialogue of nested replaceables #11119
Parameter dialog crash #11899

### Approach

Add ParentElement when create a model for replaceable parameter.
Remove condition when add parameter to window. With this condition after set redeclare component this parameter disappear from parameter`s window.

### CODE REVIEW

File: OMEdit/OMEditLIB/Element/ElementProperties.cpp [hunk 1]
```diff
@@ -1297,7 +1297,7 @@ void ElementParameters::applyFinalStartFixedAndDisplayUnitModifiers(Parameter *p
     /* Ticket #2531
      * Check if parameter is marked final.
      */
-    if ((checkFinal && pModifier->isFinal()) || (!isElementModification && pModifier->isRedeclare() && !pModifier->isReplaceable())) {
+    if (checkFinal && pModifier->isFinal()) {
```
Comment (adeas31) [inline OMEdit/OMEditLIB/Element/ElementProperties.cpp:1300]: Is this change intentional? I believe this is by mistake since these conditions are needed.

**Review by adeas31 — CHANGES_REQUESTED** [review]

File: OMEdit/OMEditLIB/Element/ElementProperties.cpp (hunk 1 above)
Comment (MaksKBV) [reply OMEdit/OMEditLIB/Element/ElementProperties.cpp:1300]: Yes, with this condition you can`t change redeclare component after you set it once and press OK. This parameter is disappear from parameter list. I will return this condion and open another issue, because this is another problem.

**Review by adeas31 — APPROVED** [review]
